### DISCUSSION TITLE
Repeated Widgets

### DISCUSSION THREAD

**Discussion by Akhad7** [TomSchimansky/CustomTkinter#909] [Q&A] [answered]
Hi, I am looking for a way to create a widget for each element of a list.

currently I managed this by coding something similar to this pseudocode:

```
for element in list:
    x+=1
    widget = customtkinter.CTk() ...
    widget.place(x,0)
```

However, I need to delete them all later on but I cannot do this because when I try doing "widget.destroy()" it only gets rid of the last placed widget.

Is there a better way to do this?

**Comment by Akascape** [comment, marked as answer]
@Akhad7 Refer this example: https://github.com/TomSchimansky/CustomTkinter/discussions/563#discussioncomment-3983861

```python
widget_list=[1, 2, 3, 4]
widget={}

for element in range(len(widget_list)):
    widget[element] = customtkinter.CTkButton(master=root)
    widget[element].grid()
```

**Reply by Akhad7** [reply]
> @Akhad7 Refer this example: https://github.com/TomSchimansky/CustomTkinter/discussions/563#discussioncomment-3983861
> 
> 
> 
> ```python
> 
> widget_list=[1, 2, 3, 4]
> 
> widget={}
> 
> 
> 
> for element in range(len(widget_list)):
> 
>     widget[element] = customtkinter.CTkButton(master=root)
> 
>     widget[element].grid()
> 
> ```

ahh, okay thank you. :))

**Reply by Akhad7** [reply]
> @Akhad7 Refer this example: [#563 (comment)](https://github.com/TomSchimansky/CustomTkinter/discussions/563#discussioncomment-3983861)
> 
> ```python
> widget_list=[1, 2, 3, 4]
> widget={}
> 
> for element in range(len(widget_list)):
>     widget[element] = customtkinter.CTkButton(master=root)
>     widget[element].grid()
> ```

Hi, I used this solution and it works however I ran into another issue.
In my code I have a command within the `widget[element]` (widget is a button) which within the callback function requires me to identify which button was pressed. If I try passing the `element` it just displays the last set value which is not useful. 

How could I identify which button was pressed in callback function?
@Akascape

**Reply by Akascape** [reply]
@Akhad7 You have to bind with **lambda command** properly.
**Example:**
```python
def clicked(x):
    print("button:"+str(x))
    
widget_list=[1, 2, 3, 4]
widget={}

for element in range(len(widget_list)):
    widget[element] = customtkinter.CTkButton(master=app, command=lambda e = element: clicked(e))
    widget[element].grid()
```
By this, you can identify the button.

**Reply by Akhad7** [reply]
> @Akhad7 You have to bind with **lambda command** properly. **Example:**
> 
> ```python
> def clicked(x):
>     print("button:"+str(x))
>     
> widget_list=[1, 2, 3, 4]
> widget={}
> 
> for element in range(len(widget_list)):
>     widget[element] = customtkinter.CTkButton(master=app, command=lambda e = element: clicked(e))
>     widget[element].grid()
> ```
> 
> By this, you can identify the button.

@Akascape 
Thank you so much! you are an absolute legend :))
